crates/net/p2p/src/full_block.rs

```diff
@@ -41,7 +42,7 @@ impl<Client> FullBlockClient<Client> {
 
 impl<Client> FullBlockClient<Client>
 where
-    Client: BodiesClient + HeadersClient + Clone,
+    Client: BlockClient + Clone,
```

Comment:
```suggestion
    Client: BlockClient,
```

Similar to this comment: https://github.com/paradigmxyz/reth/pull/10059#discussion_r1702863335